packages/headless/src/api/search/html/html-api-client.ts

```diff
@@ -0,0 +1,62 @@
+import {HtmlRequest} from './html-request';
+import {SearchAPIErrorWithStatusCode} from '../search-api-error-response';
+import {baseSearchRequest} from '../search-api-params';
+import {PlatformClient} from '../../platform-client';
+import {pickNonBaseParams, unwrapError} from '../search-api-client';
+import {findEncoding} from '../encoding-finder';
+import {TextDecoder} from 'web-encoding';
+import {Logger} from 'pino';
+import {PreprocessRequest} from '../../preprocess-request';
+
+export interface HtmlApiClient {
```

Comment:
Curious why you decided to go with the approach of an interface `HtmlApiClient` with `getHtml` exported as a pure function, instead of the pattern used for other API clients (a static class) ?

It's not a comment about necessarily changing it by the way, just curious.

Reply:
Okay, the more I'm reading, the more I'm not certain why this interface is needed.

The correct types should already  be available on `SearchApiClient` and `CaseAssistApiClient` 🤔 

That is, `CaseAssistApiClient.html()` and `SearchApiClient.html()` should already be typed properly with the return types on error and success.

And so the only the function `getHtml` present in this file and used by both client should be enough.

Reply:
The motivator behind this interface was [this interface here](https://github.com/coveo/ui-kit/pull/1701/files#diff-ada24d9636c2eb702fbde2a92daf14b08c60ac06b4212b5c4ce159d8c994b437L20) being tightly coupled to the `SearchApiClient`.

The action errored when used with a `CaseAssistEngine`, which has a `CaseAssistApiClient`.

We could have solved the error with a union: `SearchApiClient | CaseAssistApiClient`
But this union will need to continue growing as new use-cases/engines are added.

If the type is scoped html client though, then the api clients of new use-cases will just need to make sure the `html` method is implemented.

One thing though, @MehdiMitiche did you explore if you could make typescript error when passing an engine with an api-client without the `html` method implemented?

Reply:
I'm currently  exploring it, my attempts are not successful yet🧐